test/e2e/lua/dynamic_certificates.go

```diff
@@ -0,0 +1,176 @@
+			ingress, err := f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Get("foo.com", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			ingress.Spec.TLS = []extensions.IngressTLS{
+				{
+					Hosts:      []string{"foo.com"},
+					SecretName: "foo.com",
+				},
+			}
+
+			_, err = framework.CreateIngressTLSSecret(f.KubeClientSet,
+				ingress.Spec.TLS[0].Hosts,
+				ingress.Spec.TLS[0].SecretName,
+				ingress.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			resp, _, errs := gorequest.New().
+				Get(fmt.Sprintf("%s?id=certificate_only_changes", f.IngressController.HTTPURL)).
+				Set("Host", "foo.com").
+				End()
+			Expect(len(errs)).Should(BeNumerically("==", 0))
+			Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
+			_, err = f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Update(ingress)
+			Expect(err).ToNot(HaveOccurred())
+			time.Sleep(waitForLuaSync)
+
+			log, err := f.NginxLogs()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(log).ToNot(BeEmpty())
+
+			index := strings.Index(log, "id=certificate_only_changes")
+			restOfLogs := log[index:]
+
+			By("POSTing new certificates to Lua endpoint")
+			Expect(restOfLogs).To(ContainSubstring(logDynamicConfigSuccess))
+			Expect(restOfLogs).ToNot(ContainSubstring(logDynamicConfigFailure))
+
+			By("skipping Nginx reload")
+			Expect(restOfLogs).ToNot(ContainSubstring(logRequireBackendReload))
+			Expect(restOfLogs).ToNot(ContainSubstring(logBackendReloadSuccess))
+			Expect(restOfLogs).To(ContainSubstring(logSkipBackendReload))
+			Expect(restOfLogs).ToNot(ContainSubstring(logInitialConfigSync))
+		})
+	})
+
+	Context("when certificates are requested", func() {
+		It("should serve certificates dynamically from Lua", func() {
+			ingress, err := f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Get("foo.com", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			ingress.Spec.TLS = []extensions.IngressTLS{
+				{
+					Hosts:      []string{"foo.com"},
+					SecretName: "foo.com",
+				},
+			}
+
+			_, err = framework.CreateIngressTLSSecret(f.KubeClientSet,
+				ingress.Spec.TLS[0].Hosts,
+				ingress.Spec.TLS[0].SecretName,
+				ingress.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Update(ingress)
+			Expect(err).ToNot(HaveOccurred())
+			time.Sleep(waitForLuaSync)
+
+			By("checking SSL Certificate using the NGINX IP address")
+			resp, _, errs := gorequest.New().
+				Get(f.IngressController.HTTPSURL).
+				Set("Host", ingress.Spec.TLS[0].Hosts[0]).
+				TLSClientConfig(&tls.Config{
+					InsecureSkipVerify: true,
+				}).
+				End()
+
+			Expect(len(errs)).Should(BeNumerically("==", 0))
+			Expect(len(resp.TLS.PeerCertificates)).Should(BeNumerically("==", 1))
+			for _, pc := range resp.TLS.PeerCertificates {
+				Expect(pc.Issuer.CommonName).Should(Equal("default"))
```

Comment:
Where does `default` come from? Can you assert that SNI is `foo.com`? How are you making sure it is the certificate created at https://github.com/Shopify/ingress/pull/60/files#diff-b78b013bb4c03da637075b0cbe8ba522R125 that's being served?

Reply:
> but I can assert that the DNSNames contain foo.com

👍